client.go

```diff
@@ -189,16 +215,20 @@ func (c *client) mainloop(ctx context.Context, params *lookupParams) {
 	}
 
 	// Iterate through channels from listeners goroutines
-	var entries, sentEntries map[string]*ServiceEntry
-	sentEntries = make(map[string]*ServiceEntry)
+	var entries map[string]*ServiceEntry
+	c.sentEntries = make(map[string]*ServiceEntry)
+	go c.cleanupSentEntries(ctx)
```

Comment:
nit: consider doing this inline? There's no real reason to have another loop, and it saves us a lock & shared state.

Reply:
That's more than just a nit. That's a major simplification!

Reply:
nit == not really necessary but maybe nice to have. But yeah, it's cleaner now.